src/ansi-c/c_typecheck_expr.cpp

```diff
@@ -1947,8 +1947,59 @@ void c_typecheck_baset::typecheck_side_effect_function_call(
     if(symbol_table.symbols.find(identifier)==symbol_table.symbols.end())
     {
       // This is an undeclared function.
+
+      // Is it the polymorphic typed_target function ?
+      if(identifier == CPROVER_PREFIX "typed_target")
+      {
+        if(expr.arguments().size() != 1)
+        {
+          std::ostringstream error_message;
+          error_message << "expected 1 argument for "
+                        << CPROVER_PREFIX "typed_target"
+                        << " found " << expr.arguments().size();
+          throw invalid_source_file_exceptiont{
+            error_message.str(), expr.source_location()};
+        }
+
+        auto arg0 = expr.arguments().at(0);
+        typecheck_expr(arg0);
+        if(!is_assignable(arg0) || !arg0.get_bool(ID_C_lvalue))
+        {
+          std::ostringstream error_message;
+          error_message << "argument of " << CPROVER_PREFIX "typed_target"
+                        << "must be assignable";
+          throw invalid_source_file_exceptiont{
```

Comment:
You could choose to avoid the stream here as well, the string is a constant.

Reply:
fixed